public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -164,6 +177,44 @@ export const QueryAndExpressionsStep: FC<Props> = ({ editingExistingRule, onData
     [queries]
   );
 
+  const onChangeRecordingRulesQueries = useCallback(
+    (updatedQueries: AlertQuery[]) => {
+      const dataSourceSettings = getDataSourceSrv().getInstanceSettings(updatedQueries[0].datasourceUid);
+      if (!dataSourceSettings) {
+        throw new Error('The Data source has not been defined.');
+      }
+
+      //@ts-ignore
```

Comment:
you can do that instead of adding ts-ignore
```suggestion
      const expression = isPromQuery(updatedQueries[0].model) ? updatedQueries[0].model.expr : ''
```

Reply:
you have to export the `isPromQuery` function, from `rule-form.ts`

Reply:
@ts-ignore removed 🗑